pkg/R/utils.R

```diff
@@ -419,3 +419,82 @@ cleanClosure <- function(func, checkedFuncs = new.env()) {
   }
   func
 }
+
+# Common function for implmentation of zipRDD() and cartesian().
+# param
+#   x An RDD.
+#   other An RDD.
+#   operation The specified operation, "zip" or "cartesian".
+#
+# return value
+#   The result RDD.
+doZipOrCartesian <- function(x, other, operation = c("zip", "cartesian")) {
```

Comment:
I'm wondering if there is a better way to do this than have this one large function. i.e can we break this up into say `appendLength` and the second one being `mergePartitions(checkEqualLength, compactResults)`. Then `zip` can look like
```
zipRDDs <- function() {
  appendLengths()
  <call JRDD zip here>
  mergePartitions(checkLengths = T, compactResults = F)
}
```